File: src/main/java/fastcampus/backendsignature/webflux/controller/PostController.java
package fastcampus.backendsignature.webflux.controller;

import fastcampus.backendsignature.webflux.dto.PostResponse;
import fastcampus.backendsignature.webflux.service.PostService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/posts")
public class PostController {
    private final PostService postService;

    @GetMapping("/{id}")
    public Mono<PostResponse> getPost(@PathVariable(name = "id") Long id) {
        return postService.getPost(id);
    }

    @GetMapping("/search")
    public Flux<PostResponse> getPosts(@RequestParam(name = "ids") List<Long> ids) {
//        return postService.getPosts(ids);
        return postService.getPostsUsingParallel(ids);
    }
}

File: temp-api/src/main/kotlin/com/tempapi/controller/PostController.kt
package com.tempapi.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/posts")
class PostController {
    @GetMapping("/{id}")
    fun getPost(@PathVariable id: Long): Map<String, String> {
        println("id : $id")

        if (id > 9L) {
            throw Exception("존재하지 않는 게시물입니다.")
        }

        Thread.sleep(1_000)

        return mapOf(
            "id" to id.toString(),
            "contents" to "%d의 본문".format(id)
        )
    }
}